### PR TITLE
Kernel: Pull apart the mess that is CPU.h

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
There are only a few dangling things left in there, where I dont know where to put them (PAGE_SIZE for example)

### CODE REVIEW

**Comment by pheonixfirewingz** [comment]
As far as i have read this makes no functionality changes to the kernel is this just a refactoring of the code?

**Comment by Hendiadyoin1** [comment]
yes this is just refactoring, due to the previous version being a nightmare to work around

**Comment by bgianfo** [comment]
> yes this is just refactoring, due to the previous version being a nightmare to work around

🙄 maybe keep the disparaging comments to yourself, they don't add to the conversation, or your patch.

 Software's about iteration and refactoring, your beautiful design will be tomorrow's nightmare. 😄

File: Kernel/Arch/x86/common/Processor.cpp [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+    iretframe.edx = tss.edx;
+    iretframe.ecx = tss.ecx;
+    iretframe.eax = tss.eax;
+    iretframe.eflags = tss.eflags;
+    iretframe.eip = tss.eip;
+    iretframe.cs = tss.cs;
+    if (return_to_user) {
+        iretframe.userspace_esp = tss.esp;
+        iretframe.userspace_ss = tss.ss;
+    }
```
Comment (gunnarbeutner) [inline Kernel/Arch/x86/common/Processor.cpp:597]: Does having this in `x86/common` make sense considering we access IA32-specific registers?

File: Kernel/Arch/x86/x86_64/Processor.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Arch/x86/Processor.h>
+#include <Kernel/Arch/x86/TrapFrame.h>
+#include <Kernel/Process.h>
+#include <Kernel/Thread.h>
```
Comment (gunnarbeutner) [inline Kernel/Arch/x86/x86_64/Processor.cpp:10]:
```suggestion
#include <Kernel/Panic.h>
#include <Kernel/Process.h>
#include <Kernel/Thread.h>
#include <Kernel/Sections.h>
```

File: Kernel/Arch/x86/common/Processor.cpp (hunk 1 above)
Comment (Hendiadyoin1) [reply Kernel/Arch/x86/common/Processor.cpp:597]: The namems in `RegisterState` are not accurate in x86_64 and who knows if the x64 version of the ptraceregisters is up to spec,
in the short time, we might only have a 32 bit userspace, while the Kernel might be already 64 bit, which would maybe alleviate this situation or cause even more problems....
I'll move that function for now, next time I have time to program

Comment (Hendiadyoin1) [reply Kernel/Arch/x86/common/Processor.cpp:597]: Looking at it agian, its the TSS and that one got wierd on 64 bit...
so definitely moving this one

File: Kernel/Arch/x86/x86_64/Processor.cpp (hunk 2 above)
Comment (Hendiadyoin1) [reply Kernel/Arch/x86/x86_64/Processor.cpp]: Yeah, sure

**Comment by Hendiadyoin1** [comment]
@gunnarbeutner anything else?

(I needed to redo some history due to having fixed up the wrong thing)

**Pull request merged**